### ISSUE TITLE
useless_vec false positive for `VecArgs::Repeat` when element does not implement `Copy`

### ISSUE THREAD

**Issue by Marcono1234** [rust-lang/rust-clippy#11958]
### Summary

The `useless_vec` lint erroneously flags usage of `VecArgs::Repeat` where an element is repeated multiple times, and does not implement `Copy` but only `Clone`.

The `vec!` macro supports this and invokes `clone`; however arrays only seem to support `Copy` and therefore the Clippy suggestion causes a compilation error.

### Lint Name

useless_vec

### Reproducer

I tried this code:

```rust
fn main() {
    fn f(_s: &[String]) {
    }

    f(&vec!["test".to_owned(); 2]);
}
```

I saw this happen:

```
|     f(&vec!["test".to_owned(); 2]);
|       ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: you can use a slice directly: `&["test".to_owned(); 2]`
```

But when trying this Clippy suggestion the following compiler error is shown:
```
|     f(&["test".to_owned(); 2]);
|         ^^^^^^^^^^^^^^^^^ the trait `std::marker::Copy` is not implemented for `std::string::String`
|
= note: the `Copy` trait is required because this value will be copied for each element of the array
```

I expected to see this happen:
_No Clippy warning_

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-pc-windows-msvc
release: 1.74.0
LLVM version: 17.0.4
```

And also with Clippy 0.1.76 (2023-12-12 3340d49) on Rust Playground.

### Additional Labels

@rustbot label I-suggestion-causes-error
